core/src/main/java/bisq/core/api/model/PaymentAccountTypeAdapter.java

```diff
@@ -350,14 +378,61 @@ private boolean didReadTradeCurrenciesField(JsonReader in,
         return false;
     }
 
+    private Optional<List<TradeCurrency>> getReconciledTradeCurrencies(List<String> currencyCodes,
+                                                                       PaymentAccount account) {
+        if (account.hasPaymentMethodWithId(ADVANCED_CASH_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllAdvancedCashCurrencies());
+        else if (account.hasPaymentMethodWithId(AMAZON_GIFT_CARD_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllAmazonGiftCardCurrencies());
+        else if (account.hasPaymentMethodWithId(CAPITUAL_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllCapitualCurrencies());
+        else if (account.hasPaymentMethodWithId(MONEY_GRAM_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllMoneyGramCurrencies());
+        else if (account.hasPaymentMethodWithId(PAXUM_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllPaxumCurrencies());
+        else if (account.hasPaymentMethodWithId(PAYSERA_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllPayseraCurrencies());
+        else if (account.hasPaymentMethodWithId(REVOLUT_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllRevolutCurrencies());
+        /*else if (account.hasPaymentMethodWithId(SWIFT_ID))
+            return getTradeCurrenciesInList(currencyCodes, new ArrayList<>(getAllSortedFiatCurrencies()));*/
+        else if (account.hasPaymentMethodWithId(TRANSFERWISE_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllTransferwiseCurrencies());
+        else if (account.hasPaymentMethodWithId(UPHOLD_ID))
+            return getTradeCurrenciesInList(currencyCodes, getAllUpholdCurrencies());
+        else
+            return Optional.empty();
+    }
+
+    private boolean didReadSelectedTradeCurrencyField(JsonReader in,
+                                                      PaymentAccount account,
+                                                      String fieldName) {
+        if (fieldName.equals("selectedTradeCurrency")) {
+            String fieldValue = nextStringOrNull(in);
+            if (fieldValue != null && !fieldValue.isEmpty()) {
+                Optional<TradeCurrency> tradeCurrency = getTradeCurrency(fieldValue.toUpperCase());
+                if (tradeCurrency.isPresent()) {
+                    account.setSelectedTradeCurrency(tradeCurrency.get());
+                } else {
```

Comment:
It does read the field, but should it return true even when it's not defined?

Reply:
If the json field is not defined, it can't be read.   If it is defined, it is always read.

`if (fieldName.equals("selectedTradeCurrency")) ` fails, this method needs to return false so the calling read loop does not  `continue` to the next json field.  

Look at `PaymentAccount read(JsonReader in)`  [Line 215-254]:

If `didReadSelectedTradeCurrencyField()` returns false, the `currentFieldName` might be read in `didReadCommonField()` or `didReadCountryField()`. 

If `didReadSelectedTradeCurrencyField()` returns true, the `currentFieldName` cannot be read in `didReadCommonField()` or `didReadCountryField()`.